UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs

```diff
@@ -94,7 +94,7 @@ public override string ToString()
 	{
 		return Equals(Invalid)
 			? "[Invalid matrix]"
-			: $"[({Id}){GameObject.name},offset={Offset},pivot={MatrixMove?.Pivot},state={MatrixMove?.ServerState},netId={NetID}]";
+			: $"[({Id}){GameObject?.name ?? "MatrixInfo"},offset={Offset},pivot={MatrixMove?.Pivot},state={MatrixMove?.ServerState},netId={NetID}]";
```

Comment:
NullReferenceException: Object reference not set to an instance of an object
MatrixInfo.ToString () (at /github/workspace/UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs:95)
VVUIElementHandler.Serialise (System.Object InObject, System.Type TypeOf) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/ElementDisplay/VariableViewerUIFunctionHandler.cs:157)
VariableViewerNetworking.ProcessBook (Librarian+Book _book) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs:309)
BookNetMessage.Send (Librarian+Book _book, UnityEngine.GameObject ToWho) (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/VariableViewer/BookNetMessage.cs:22)
VariableViewer.SendBookToClient (Librarian+Book Book, UnityEngine.GameObject ToWho) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:117)
VariableViewer.RequestSendBook (System.UInt64 BookID, UnityEngine.GameObject ByWho) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:216)
OpenBookIDNetMessage.ValidateAdmin () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/VariableViewer/OpenBookIDNetMessage.cs:21)
OpenBookIDNetMessage.Process () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/VariableViewer/OpenBookIDNetMessage.cs:14)
GameMessageBase.Process (Mirror.NetworkConnection sentBy) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:25)
ClientMessage.Process (Mirror.NetworkConnection sentBy) (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/ClientMessage.cs:14)
GameMessageBase.PreProcess (Mirror.NetworkConnection sentBy, GameMessageBase b) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:18)
Mirror.MessagePacker+<>c__DisplayClass6_0`2[T,C].<MessageHandler>b__0 (Mirror.NetworkConnection conn, Mirror.NetworkR

Reply:
using ``?`` on a type derived from UnityEngine.Object should be avoided. 

null checking with GameObject and Matrix should be done using the regular ``== null`` expression.

Reply:
corp-0 is there a reason for avoiding it?

Reply:
https://github.com/JetBrains/resharper-unity/wiki/Possible-unintended-bypass-of-lifetime-check-of-underlying-Unity-engine-object